File: core/src/main/java/fr/sncf/osrd/new_infra_state/api/SignalizationStateView.java
package fr.sncf.osrd.new_infra_state.api;

import fr.sncf.osrd.new_infra.api.signaling.Signal;
import fr.sncf.osrd.new_infra.api.signaling.SignalState;

public interface SignalizationStateView {
    /** Returns the state of the given signal */
    <T extends SignalState> T getSignalState(Signal<T> signal);
}

File: core/src/main/kotlin/fr/sncf/osrd/dyn_infra/api/SignalizationStateView.kt
package fr.sncf.osrd.dyn_infra.api

import fr.sncf.osrd.infra.api.signaling.SignalState
import fr.sncf.osrd.infra.api.signaling.Signal

interface SignalizationStateView {
    /** Returns the state of the given signal  */
    fun <T : SignalState> getSignalState(signal: Signal<T>): T
}